### ISSUE TITLE
Problem deploying a Simple EV3-G program on LMS-Compact when using Wi-Fi

### ISSUE THREAD

**Issue by jabrena** [ev3dev/lms2012-compat#42]
Hi @dlech,

Great feature for the whole Mindstoms community.

I read your post and I have tried to test LMS-Compact but I receive the following error from EV3-G IDE:

![](https://raw.githubusercontent.com/ilovenetruinos/ev3g-odometry/feature/ODOM_13/docs/lms/lms-compact-wifi-deployment-error.png)

The connection with the Laptop was using a Wifi connection but after some failed deployment, I tried to disconnect the wifi to use later a USB connection but the IDE didn´t detect the brick with LMS process running on EV3Dev. I had to remove the batteries to shut down the brick. The UI didn´t respond.

This is the messages which I received in the console:

``` bash
[sudo] password for robot: 
Failed to find system power supply

** (process:533): WARNING **: Failed to open /dev/hidg0: No such file or directory
Could not find 'PCM' volume control
Could not find 'Beep' volume control

** (process:533): WARNING **: Error setting property 'Discoverable' on interface org.bluez.Adapter1: GDBus.Error:org.bluez.Error.Failed: Not Powered (g-io-error-quark, 36)
``` 

Happy new year.

Cheers

Juan Antonio

**Comment by laurensvalk** [comment]
I am getting this occasionally too, especially often when the robot (on WiFi) is farther from the router.

**Comment by laurensvalk** [comment]
I've not had this when using USB for transferring programs (and using BT for starting sudo lms2012).

**Comment by dlech** [comment]
Hmm... a flaky Wi-Fi connection will be hard to reproduce. If anyone comes up with a reliable way to reproduce this, I will have a look again, but until then, I'm afraid there is not much I can do.

**Comment by dlech** [comment]
Ooo... not as hard as I though... http://nshipster.com/network-link-conditioner/

**Comment by dlech** [comment]
I was able to reproduce this error by using a network link conditioner. I was also able to reproduce it using the official EV3 firmware (v1.09). So, it looks like this is really a problem in the desktop software. The desktop software can't handle a bad network connection. In fact, when I tried to connect to a brick running the official firmware while using the network link conditioner, it crashed the desktop software.
